src/epoch.rs

```diff
@@ -29,41 +29,45 @@ impl Epoch {
 
     /// Attempts to advance the global epoch.
     ///
-    /// The global epoch can advance only if all currently pinned mutators have been pinned in the
-    /// current epoch.
+    /// The global epoch can advance only if all currently pinned participants have been pinned in
+    /// the current epoch.
     ///
     /// Returns the current global epoch.
     ///
     /// `try_advance()` is annotated `#[cold]` because it is rarely called.
     #[cold]
-    pub fn try_advance<'scope>(&self, registries: &List<LocalEpoch>, scope: &Scope) -> usize {
+    pub fn try_advance<'scope>(
+        &'scope self,
+        registries: &'scope List<LocalEpoch>,
+        scope: &'scope Scope,
```

Comment:
Why are all three lifetimes bound like this?

Note that the lifetimes here are covariant so this doesn't affect anything - any `&'scope Foo` reference is a subtype of a reference with an even smaller lifetime: `&'very_small Foo`. So whether you bind these lifetimes together or don't, it doesn't matter.